### ISSUE TITLE
ProjectDependenciesCommandFactory cannot find command in desktop .NET class library project

### ISSUE THREAD

**Issue by natemcmaster** [dotnet/cli#2919]
## Steps to reproduce
Create a sample project using dotnet-ef
```json
{
  "version": "1.0.0-*",
  "buildOptions": {
    "emitEntryPoint": false
  },
  "dependencies": {
    "Microsoft.EntityFrameworkCore.Sqlite": "1.0.0-*",
    "Microsoft.EntityFrameworkCore.Tools": {
      "type": "build",
      "version": "1.0.0-*"
    }
  },
  "frameworks": {
    "net451":{}
  },
  "tools": {
    "Microsoft.EntityFrameworkCore.Tools": {
      "imports": [
        "portable-net451+win8"
      ],
      "version": "1.0.0-*"
    }
  }
}
```
```
dotnet restore
dotnet ef migrations list --verbose
```

## Expected  behavior
dotnet-ef.dll (in Microsoft.EntityFrameworkCore.Tools) should dispatch via ProjectDependenciesCommandFactory to Microsoft.EntityFrameworkCore.Tools.Cli.exe (a project dependency) and list all migrations in the project

## Actual behavior
```
No executable found matching command "Microsoft.EntityFrameworkCore.Tools.Cli"
```

`ProjectDependenciesCommandFactory` tries to find a command in `bin/Debug/net451/win7-x64`, but the command is actually in `bin/Debug/net451/`

## Environment data
`dotnet --info` output:
```
.NET Command Line Tools (1.0.0-rc2-002673)

Product Information:
 Version:     1.0.0-rc2-002673
 Commit Sha:  c0aeb91d61

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.10586
 OS Platform: Windows
 RID:         win10-x64
```
```
"Microsoft.DotNet.Cli.Utils/1.0.0-rc2-002673"
```

**Comment by brthor** [comment]
@natemcmaster 

If you put
```
runtimes: {
    "win7-x64" : {}
}
```

in your project.json does this work?

**Comment by Sridhar-MS** [comment]
@brthor I was about to comment the same thing :)

**Comment by Sridhar-MS** [comment]
I tried with the runtimes section but I get a different error 

`Could not invoke this command on the project. Check that the version of Microsoft.EntityFrameworkCore.Tools in "tools" and "dependencies" are the same`

**Comment by natemcmaster** [comment]
It's the same error. Build output is still in `bin/Debug/net451/` :-/

**Comment by brthor** [comment]
@natemcmaster did you rerun restore?

**Comment by brthor** [comment]
Setting up a local repro

**Comment by natemcmaster** [comment]
@brthor yes. Deleted bin/ and obj/ and rebuilt too.

**Comment by Sridhar-MS** [comment]
@natemcmaster is this the correct Nuget.config?

```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <!--To inherit the global NuGet package sources remove the <clear/> line below -->
    <clear />
    <add key="aspnet-release" value="https://www.myget.org/F/aspnetrelease/api/v3/index.json" />
    <add key="api.nuget.org" value="https://api.nuget.org/v3/index.json" />
  </packageSources>
</configuration>
```

**Comment by natemcmaster** [comment]
@Sridhar-MS yes, that's the feed.

**Comment by brthor** [comment]
If you make `emitEntryPoint` true, this works.

Looking into fixes for emitEntryPoint False.

**Comment by mikeandersun** [comment]
`dotnet build` produces this error :
`An item with the same key has already been added. Key: Microsoft.DotNet.ProjectM
odel.ProjectContextBuilder+LibraryKey`

**Comment by brthor** [comment]
@mikeandersun

Is this build break from the project above? I am not seeing this.

**Comment by natemcmaster** [comment]
This is related to https://github.com/dotnet/cli/issues/2645

**Comment by piotrpMSFT** [comment]
Per #2645, running dependency commands for libraries is currently unsupported. We're fixing this issue by providing a cleaner error message.

**Comment by piotrpMSFT** [comment]
Fixed by #2933
